### PR TITLE
Don't let ws cookie get propagated to other workspaces

### PR DESCRIPTION
**Pull request by filiptronicek**
## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@filiptronicek: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/user/user-controller.ts [hunk 1]
```diff
@@ -280,13 +280,15 @@ export class UserController {
...
+
                 res.cookie(name, token, {
                     path: "/",
                     httpOnly: true,
                     secure: true,
                     maxAge: 1000 * 60 * 60 * 24 * 1, // 1 day
-                    sameSite: "lax", // default: true. "Lax" needed for cookie to work in the workspace domain.
-                    domain: `.${this.config.hostUrl.url.host}`,
+                    sameSite: "strict",
+                    domain: workspaceInstanceDomain,
```
Comment (akosyakov) [inline components/server/src/user/user-controller.ts:291]: how port locations are going to work? webviews?

**Pull request closed without merging**